File: Leet 2785/solution.py
class Solution:
    def sortVowels(self, s: str) -> str:
        result: list[str] = [c for c in s]


        vowels: list[str] = []
        idxs: list[int] = []
        for idx, ch in enumerate(s):
            if ch in "aeiouAEIOU":
                vowels.append(ch)
                idxs.append(idx)

        vowels.sort()

        for i, v in zip(idxs, vowels):
            result[i] = v

        return "".join(result)
    

def main() -> None:
    print("2785. Sort Vowels in a String")

    sol = Solution()
    
    result: str = sol.sortVowels(s="lEetcOde")
    print(result)


if __name__ == "__main__":
    main()

File: Leet 2785/solution.cs
public class Solution
{
    public string SortVowels(string s)
    {
        List<char> chars = [];
        List<int> index = [];

        const string vowels = "aeiouAEIOU";
        for (int i = 0; i < s.Length; i++)
        {
            if (vowels.Contains(s[i]))
            {
                chars.Add(s[i]);
                index.Add(i);
            }
        }

        chars.Sort();

        StringBuilder sb = new(s);
        for (int i = 0; i < chars.Count; i++)
        {
            sb[index[i]] = chars[i];
        }
        return sb.ToString();
    }
}